src/HotChocolate/Core/test/Types.Tests/Types/Relay/PolymorphicGlobalIdsTypeInterceptor.cs

```diff
@@ -0,0 +1,160 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using HotChocolate.Configuration;
+using HotChocolate.Internal;
+using HotChocolate.Types.Descriptors;
+using HotChocolate.Types.Descriptors.Definitions;
+using Microsoft.Extensions.DependencyInjection;
+
+#nullable enable
+
+namespace HotChocolate.Types.Relay
+{
+    internal sealed class PolymorphicGlobalIdsTypeInterceptor : TypeInterceptor
```

Comment:
I'll probably remove this and just unit test the bits of code I added into `GlobalIdInputValueFormatter`, but for now this was my end goal, so just whacked it in.

I'll move this poly id stuff into a new nuget package for the community. Name suggestions welcome!